SampleBase/src/SampleApp.cpp

```diff
@@ -267,7 +274,7 @@ void SampleApp::InitializeDiligentEngine(const NativeWindow* pWindow)
                                     "or required features may not be supported by this GPU/driver/OS version.");
             }
 
-            if (pWindow != nullptr)
+            if (pWindow != nullptr && !m_bOffscreenRendering)
```

Comment:
`m_bOffscreenRendering` flag should not be needed.
Instead, [native app](https://github.com/DiligentGraphics/DiligentTools/blob/master/NativeApp/src/Win32/WinMain.cpp) should handle the command line argument and do not create the window.
Then, if `pWindow` is null, the sample app will create an off-screen swap chain.